src/troute-routing/troute/routing/fast_reach/mc_reach.pyx

```diff
@@ -621,6 +624,21 @@ cpdef object compute_network_structured(
                     lastobs_values[gage_i],
                     gage_i == da_check_gage,
                 )
+                 # Ensure we're within bounds before accessing the arrays
+                if timestep < usgs_values.shape[1]:
+                    usgs_value = usgs_values[gage_i, timestep]
+                    usgs_values_list.append(usgs_value)
+
+                if timestep < flowveldepth.shape[1]:
+                    flowveldepth_value = flowveldepth[usgs_position_i, timestep]
+                    flowveldepth_list.append(flowveldepth_value)
+
+                
+                da_buf0_value = da_buf[0]
```

Comment:
For clarity, df_buf0_value can be replaced by data_assim_flow_value.

Reply:
Done